From 75216a8178fb89835a0099efc28f8ebed5af4e8e Mon Sep 17 00:00:00 2001
From: Takashi Sakamoto <o-takashi@sakamocchi.jp>
Date: Sun, 17 Sep 2023 11:52:28 +0900
Subject: [PATCH] hwdb: ieee1394-unit-function: add Weiss Engineering INT203
 entry with older firmware

Michele Perrone reported that the model ID for INT203 with initial firmware
is 0x00000a.

* https://lore.kernel.org/alsa-devel/a5b63c37-7f6f-46f3-a87f-e4e7c0c9b5c8@weiss.ch/

This commit fulfills the entry corresponding to it as well as supplemental
comment denoting to it.

(cherry picked from commit 618edc0f566bf36843b32c9beb3dfa64db2371cf)

Related: RHEL-5950
---
 hwdb.d/80-ieee1394-unit-function.hwdb | 14 +++++++++++++-
 1 file changed, 13 insertions(+), 1 deletion(-)

diff --git a/hwdb.d/80-ieee1394-unit-function.hwdb b/hwdb.d/80-ieee1394-unit-function.hwdb
index 503f5bbc14..bca410495a 100644
--- a/hwdb.d/80-ieee1394-unit-function.hwdb
+++ b/hwdb.d/80-ieee1394-unit-function.hwdb
@@ -912,10 +912,14 @@ ieee1394:ven00001C6Amo00000004sp00001C6Aver00000001
  ID_MODEL_FROM_DATABASE=AFI1
  IEEE1394_UNIT_FUNCTION_AUDIO=1
 
+# Starting from the middle of life time of INT203, the same firmware as
+# INT202 is compatible. The firmware dynamically switches its functionality
+# based on the state of internal jumper pins. Additionally, the design of
+# FireWire option card for DAC1 is something like embedded-INT203.
 ieee1394:node:ven0x001c6amo0x000006units0x001c6a:0x000001
 ieee1394:ven00001C6Amo00000006sp00001C6Aver00000001
  ID_VENDOR_FROM_DATABASE=Weiss Engineering
- ID_MODEL_FROM_DATABASE=INT202
+ ID_MODEL_FROM_DATABASE=INT202/INT203/DAC1
  IEEE1394_UNIT_FUNCTION_AUDIO=1
 
 ieee1394:node:ven0x001c6amo0x000007units0x001c6a:0x000001
@@ -930,6 +934,14 @@ ieee1394:ven00001C6Amo00000008sp00001C6Aver00000001
  ID_MODEL_FROM_DATABASE=DAC202/Maya
  IEEE1394_UNIT_FUNCTION_AUDIO=1
 
+# The older version of firmware for INT203. Additionally, the design of
+# FireWire option card for DAC1 is something like embedded-INT203.
+ieee1394:node:ven0x001c6amo0x00000aunits0x001c6a:0x000001
+ieee1394:ven00001C6Amo0000000Asp00001C6Aver00000001
+ ID_VENDOR_FROM_DATABASE=Weiss Engineering
+ ID_MODEL_FROM_DATABASE=INT203/DAC1
+ IEEE1394_UNIT_FUNCTION_AUDIO=1
+
 #
 # Digidesign Digi00x family.
 #
